langchain/src/experimental/openai_files/index.ts

```diff
@@ -0,0 +1,86 @@
+import { OpenAI as OpenAIClient } from "openai";
+import fs from "node:fs";
+import { Serializable } from "../../load/serializable.js";
+
+export type OpenAIFilesCreate = {
+  file: fs.ReadStream;
+  purpose: "assistants" | "fine-tune";
+};
+
+export class OpenAIFiles extends Serializable {
+  lc_namespace = ["langchain", "experimental", "open_ai_files"];
+
+  /**
+   * Upload file
+   * Upload a file that can be used across various endpoints. The size of all the files uploaded by one organization can be up to 100 GB.
+   *
+   * The size of individual files can be a maximum of 512 MB. See the Assistants Tools guide to learn more about the types of files supported. The Fine-tuning API only supports .jsonl files.
+   *
+   * @link {https://platform.openai.com/docs/api-reference/files/create}
+   * @param {fs.ReadStream} file
+   * @param {"assistants" | "fine-tune"} purpose
+   * @returns {Promise<OpenAIClient.Files.FileObject>}
+   */
+  static async createFile({ file, purpose }: OpenAIFilesCreate) {
+    const oaiClient = new OpenAIClient();
```

Comment:
We should probably create a `client` instance in a constructor and have it as an instance property so that people can pass custom OpenAI keys/organizations if needed